app/javascript/src/awards.js

```diff
@@ -7,18 +7,51 @@ document.addEventListener("turbolinks:load", () => {
       post_id: postId,
       award_id: awardId,
     };
-    
+
     $.ajax({
       type: "POST",
       url: "/reward/",
       data: postData,
-      success: function(){
-        $('#post-awards').append(`<img src="${self.src}" class="post-award"/>`);
-        $("#awardsSelectModal").modal("hide");
+      success: function(response){
+        if (response["bought"]) {
+          $('#post-awards').append(`<img src="${self.src}" class="post-award"/>`);
+          $("#awardsSelectModal").modal("hide");
+        }
       },
       error: function(){
         console.error("Error occured in reward process!");
       }
     });
   });
+
+  var originalValue;
+  $(document).on("focusout", ".js-value-input", function() {
+    $self = $(this);
+    let value = $self.val();
+    let awardId = $self.data("id");
+    value = (value === "") ? "1" : value;
```

Comment:
Es ne ļoti saprotu visas šis manipulacijas. Ja tu gribi parliecinaties kā lietotais vienmer padot talak number, uzliec input laukam iekš html type="number"

Reply:
vins parbauda vai nav tukss lauks. input type jau ir number

Reply:
Jā pareizi saprotu tad gadijumā jā vertiba ir tukšam tad viņai jabūt "1"?
Nevar šeit atstat tikai:
 ```javascript
   $.ajax({
      type: "PUT",
      url: `/admin/awards/${awardId}`,
      data: {
        value: "" || "1"
      },
      success: function() {
        $self.replaceWith(`<i class="js-award-value" data-id="${awardId}">${value}</i>`)
      },
      error: function() {
        $self.replaceWith(`<i class="js-award-value" data-id="${awardId}">${originalValue}</i>`)
      }
    })
```

Reply:
rather turetu vinu mainigaja, since vinu izmantoju pectam success